server/src/main/java/org/opensearch/telemetry/tracing/SpanBuilder.java

```diff
@@ -75,7 +76,12 @@ public static SpanCreationContext from(String spanName, String nodeId, Replicate
     }
 
     private static String createSpanName(HttpRequest httpRequest) {
-        return httpRequest.method().name() + SEPARATOR + httpRequest.uri();
+        try {
+            URI uri = new URI(httpRequest.uri());
```

Comment:
We'll be creating `URI` at least twice every time (typical usage below):

```
    public static SpanCreationContext from(HttpRequest request) {
        return SpanCreationContext.server().name(createSpanName(request)).attributes(buildSpanAttributes(request));
    }
```

I think we could borrow a very simple `RestRequest#path` implementation to split uri to path + query string without creating tons of URI class instances.